src/lib/api/api.go

```diff
@@ -339,9 +340,10 @@ func (a *API) AddRouteToMonitoring(w http.ResponseWriter, r *http.Request) {
 		),
 	)
 	a.reloadConfigURLs <- struct{}{}
+	// wait for reload to finish
+	<-a.receiveFinishSignal
```

Comment:
The reason this was kept in 344 is, routes can take a lot of time to reload. We do not have to wait for that to happen. If we wait, the only reason would have been an error possibility which I don't see happening (since the reloading goroutine does not return any error as there isn't anything to cause an error).

Reply:
Do you think we can drop the `done` channel?